src/osal/lv_linux.c

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file lv_linux.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_os.h"
+
+#if LV_USE_OS != LV_OS_NONE && defined(__linux__)
+
+#include "../misc/lv_log.h"
+#include <stdio.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#define LV_UPTIME_MONITOR_FILE         "/proc/stat"
+
+#define LV_PROC_STAT_VAR_FORMAT        " %" PRIu32
+#define LV_PROC_STAT_IGNORE_VAR_FORMAT " %*" PRIu32
+#define LV_PROC_STAT_PARAMS_LEN        7
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef union {
+    struct {
+        /*
+         *  We ignore the iowait column as it's not reliable
+         *  We ignore the guest and guest_nice columns because they're accounted
+         *   for in user and nice respectively
+         */
+        uint32_t user, nice, system, idle, /*iowait,*/ irq, softirq,
+                 steal /*, guest, guest_nice*/;
+    } fields;
+    uint32_t buffer[LV_PROC_STAT_PARAMS_LEN];
+} lv_proc_stat_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static lv_result_t lv_read_proc_stat(lv_proc_stat_t * result);
+static uint32_t lv_proc_stat_get_total(const lv_proc_stat_t * p);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+static lv_proc_stat_t last_proc_stat;
```

Comment:
It should go to `lv_global.h`. In this file you can easily reference it with `#define last_proc_stat LV_GLOBAL_DEFAULT()->linux_last_proc_stat`